pandas/tests/groupby/test_groupby_dropna.py

```diff
@@ -454,3 +456,22 @@ def test_no_sort_keep_na(values, dtype, test_series):
         # TODO: Slicing reorders categories?
         expected.index = expected.index.reorder_categories(["y", "x"])
     tm.assert_equal(result, expected)
+
+
+@pytest.mark.parametrize("test_series", [True, False])
+@pytest.mark.parametrize("dtype", [object, None])
+def test_null_is_null_for_dtype(
+    sort, dtype, unique_nulls_fixture, unique_nulls_fixture2, test_series
+):
+    # GH#48506 - groups should always result in using the null for the dtype
+    df = pd.DataFrame({"a": [1, 2]})
+    groups = pd.Series([unique_nulls_fixture, unique_nulls_fixture2], dtype=dtype)
```

Comment:
Would it be good to include `pd.NA` here (which isn't in unique_nulls_fixture)?

Reply:
Absolutely - thanks, I didn't realize.